### ISSUE TITLE
LC0068 triggered in test codeunits despite TestPermissions = Disabled

### ISSUE THREAD

**Issue by frottke** [StefanMaron/BusinessCentral.LinterCop#1201]
I appreciate the work that goes into this project. In that context, I noticed a behavior I would like to ask about. It appears that LC0068 does not respect the TestPermissions setting in test codeunits when its value is Disabled. 

In the example below, the call to SalesLine.Modify(false) is still flagged by LC0068, even though the test codeunit explicitly sets TestPermissions = Disabled.

```
codeunit 99999 "mycodeunit"
{
    Subtype = Test;
    TestPermissions = Disabled;
    //Permissions = tabledata "Sales Line" = m;

    local procedure myprocedure(var SalesLine: Record "Sales Line")
    begin
        SalesLine.Modify(false); //LC0068 is triggered but TestPermissions = Disabled is set
    end;
}
```

Expected behavior

When a test codeunit uses TestPermissions = Disabled, analyzers should not report missing or insufficient permissions for operations inside that codeunit.

Actual behavior

LC0068 is reported on SalesLine.Modify(false) despite the TestPermissions = Disabled setting.

Questions

Is this behavior intentional?
If so, what is the reasoning behind ignoring TestPermissions for LC0068 in test codeunits?

Additional context

The documentation suggests that TestPermissions = Disabled suppresses permission checks at runtime, which would imply that analyzer warnings should also be suppressed for such codeunits. Clarification would help determine whether this is a bug, a design decision, or a limitation of the analyzer.

**Comment by finnpedersenkazes** [comment]
I see it too. Maybe it is new. I see it in version 28.0. I like to have the same codeAnalyzers for my test app as for the app, but maybe it is not possible.

> **TestPermissions = Disabled**
> Setting this value will exclude any change of the permission execution context and all tests will be executed using SUPER.

Maybe **LC0068** has nothing to do with the [TestPermissions Property](https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/properties/devenv-testpermissions-property)

**Alternative Solution - Disable the warning**

We place these lines in the top and bottom of the test Codeunits. 

```
#pragma warning disable LC0068

#pragma warning restore LC0068
```

**Alternative Solution - Disable via special ruleset file for test apps**

```json
{
    "id": "LC0068",
    "action": "None",
    "justification": "Missing permissions to access tabledata (inform user)."
},
```

**Comment by Arthurvdv** [comment]
Hi, over the past months I’ve been putting a lot of time and care into the successor of BusinessCentral.LinterCop: ALCops (https://alcops.dev/). Because of that, this repository is no longer actively maintained.

I’m closing this issue, as it won’t be handled here anymore. If this is still relevant for you, please have a look at ALCops/Analyzers (https://github.com/ALCops/Analyzers).

To make the transition easier, you can also follow the LinterCop Migration Guide: https://alcops.dev/docs/lintercop-migration/

If you have any questions or run into issues during the transition, feel free to reach out through ALCops. Happy to help 🙂
